lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php

```diff
@@ -197,8 +197,16 @@ private static function get_inner_blocks_from_navigation_post( $attributes ) {
 			// it encounters whitespace. This code strips it.
 			$compacted_blocks = gutenberg_block_core_navigation_filter_out_empty_blocks( $parsed_blocks );
 
+			// Run Block Hooks algorithm to inject hooked blocks.
+			$markup = gutenberg_hook_first_last_children_into_core_navigation( $compacted_blocks, $navigation_post->ID );
```

Comment:
Shouldn't we aim to change the block to work with block hooks rather than adding specialised code in the block for block hooks case?

Reply:
The crux is that the Navigation block isn’t an actual container block — which doesn’t stop most people (myself included) of wanting to insert another block as its last child. But since the Navigation block is really using fairly specialized code in its render method, it’s this code that we need to adapt to also insert hooked blocks.

(The same is true for a number of other blocks — Template Part, Post Content, etc. The solution we’re developing might be somewhat scalable to a number of these.)

Once you try, you notice pretty quickly that the reason that it doesn’t work right OOTB with Block Hooks the way other blocks do is that specialized code, and there isn’t really any other way to make it work (barring standardization — i.e. a complete overhaul — of blocks like Navigation et al.)